exporters/common/src/main/java/io/opentelemetry/exporter/internal/TlsUtil.java

```diff
@@ -83,9 +99,8 @@ public static X509KeyManager keyManager(byte[] privateKeyPem, byte[] certificate
     try {
       KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
       ks.load(null);
-      KeyFactory factory = KeyFactory.getInstance("RSA");
       PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(decodePem(privateKeyPem));
-      PrivateKey key = factory.generatePrivate(keySpec);
+      PrivateKey key = generatePrivateKey(keySpec, SUPPORTED_KEY_FACTORIES);
```

Comment:
Maybe we can try to get the key type from `chain`. Can we move this line to after the 
```
while (is.available() > 0) {
        chain.add(cf.generateCertificate(is));
}
```

Reply:
```
String algId = chain[0].algId
```
This code doesn't compile for me, since `chain` is a list rather than an array. But even with `chain.get(0)`, I don't see anyway to access the algorithm from the [Certificate](https://docs.oracle.com/javase/8/docs/api/java/security/cert/Certificate.html) class.

Reply:
How about this? 
```
((X509Certificate) chain.get(0)).getSigAlgName()
```

Reply:
The javadoc for `getSigAlgName()` gives the example `SHA256withRSA` which may not be what we want, is it?

You can get the algorithm from the cert's public key with `cert.getPublicKey().getAlgorithm()`. The example in that javdoc gives `DSA` and links to the [standard names](https://docs.oracle.com/javase/9/docs/specs/security/standard-names.html#standard-names) which may be more in line with what we want.

Reply:
@breedx-splk You are right. The `chain.get(0).getPublicKey().getAlgorithm()` is more make sense. 

@jack-berg  What do you think?

Reply:
The certificate is parsed from the `certificatePem` argument, while the key is parsed from the `privateKeyPem` argument. Feels strange to use a field from `certificatePem` to inform the parsing strategy for `privateKeyPem`. 

Is there any reason the certificate from `certificatePem` would have a different algorithm than `privateKeyPem`?

Is it safe to always look at the algorithm from `chain.get(0)`?

Not sure of the answer to these questions. I do know that the strategy in this PR of `try / catch` doesn't have to answer those questions, and while not ideal, parsing certificates isn't something that is done very often so its a reasonable fallback strategy.

Reply:
Sorry, I'm not a certificate expert, so can't give you an accurate answer about your concern.  
For my understanding, certificate and key always a pair. At least, they should have the same algorithm.
And a valid certificate at least have one chain even though it is from a root CA, so `chain.get(0)` should be safe. But 
a certificate should contain a serial chain, so not sure which chain we need to use. And we can also add a `try/catch` for it to prevent some invalid certificate.

Reply:
Hi @jack-berg, how can we make decision for this point?

Reply:
I don't have a strong opinion on this. I think the try / catch approach is a code smell but will be a dependable strategy here, without performance implications as this doesn't run very often. On the other hand, extracting the algorithm from the certificate is more sound programing, but may run into sharp edges based off the `certifcatePem` and `privateKeyPem` being different arguments, and `certificatePem` being a chain. 

Slight preference for the try / catch approach unless someone with expertise in certificates can set the record straight.

Reply:
@jack-berg I'm happy to go ahead with your `try/catch` solutions.